store/src/java/com/zimbra/cs/session/Session.java

```diff
@@ -166,13 +167,13 @@ public Session register() throws ServiceException {
      *
      * @see #isMailboxListener()
      * @see #isRegisteredInCache() */
-    public Session unregister() {
+    public Session unregister(MailboxLock l) {
         // locking order is always Mailbox then Session
         MailboxStore mboxStore = mailbox;
         if (null != mboxStore) {
             if (mboxStore instanceof Mailbox) {
                 Mailbox mbox = (Mailbox)mboxStore;
-                assert(mbox.lock.isWriteLockedByCurrentThread() || !Thread.holdsLock(this));
+                assert(l.isWriteLockedByCurrentThread() || !Thread.holdsLock(this));
```

Comment:
_TODO_ need to figure out what to do with these calls to `Thread.holdsLock(this)`